core/node/node_framework/src/implementations/layers/base_token/base_token_ratio_persister.rs

```diff
@@ -93,7 +87,11 @@ impl WiringLayer for BaseTokenRatioPersisterLayer {
 
                 let signing_client = PKSigningClient::new_raw(
                     tms_private_key.clone(),
-                    self.contracts_config.diamond_proxy_addr,
+                    input
+                        .l1_contracts_resource
+                        .0
+                        .chain_contracts_config
```

Comment:
Nit: You can extract a reference to this config to a local var to not repeat access to it in 3 places.